ui/src/configuration/components/RandomLabelColor.tsx

```diff
@@ -0,0 +1,45 @@
+import React, {Component} from 'react'
+import _ from 'lodash'
+
+import {PRESET_LABEL_COLORS} from 'src/configuration/constants/LabelColors'
+import {IconFont} from 'src/clockface'
+
+// Styles
+import 'src/configuration/components/RandomLabelColor.scss'
+
+interface Props {
+  colorHex: string
+  onClick: (newRandomHex: string) => void
+}
+
+export default class RandomLabelColorButton extends Component<Props> {
+  public componentDidMount() {
+    this.handleClick()
```

Comment:
Using the mount of the component to eventually provide data from higher up in the render tree feels a bit funky. Can the color be generated for the first time outside of this component and then passed in?